src/acceptance-test/java/org/zalando/nakadi/webservice/ConsumerLimitingCleaningServiceAT.java

```diff
@@ -0,0 +1,118 @@
+package org.zalando.nakadi.webservice;
+
+import com.fasterxml.jackson.databind.ObjectMapper;
+import org.apache.curator.framework.CuratorFramework;
+import org.apache.zookeeper.KeeperException;
+import org.joda.time.DateTime;
+import org.joda.time.DateTimeZone;
+import org.junit.After;
+import org.junit.Before;
+import org.junit.Test;
+import org.mockito.Mockito;
+import org.zalando.nakadi.config.JsonConfig;
+import org.zalando.nakadi.repository.zookeeper.ZooKeeperHolder;
+import org.zalando.nakadi.repository.zookeeper.ZooKeeperLockFactory;
+import org.zalando.nakadi.service.ConsumerLimitingCleaningService;
+import org.zalando.nakadi.service.ConsumerLimitingService;
+import org.zalando.nakadi.webservice.utils.ZookeeperTestUtils;
+
+import java.nio.charset.Charset;
+
+import static org.hamcrest.CoreMatchers.not;
+import static org.hamcrest.CoreMatchers.nullValue;
+import static org.hamcrest.MatcherAssert.assertThat;
+import static org.mockito.Mockito.when;
+
+public class ConsumerLimitingCleaningServiceAT extends BaseAT {
+
+    private static final CuratorFramework CURATOR = ZookeeperTestUtils.createCurator(ZOOKEEPER_URL);
+
+    private ConsumerLimitingCleaningService cleaningService;
+    private ObjectMapper objectMapper;
+
+    public ConsumerLimitingCleaningServiceAT() {
+        final ZooKeeperHolder zkHolder = Mockito.mock(ZooKeeperHolder.class);
+        when(zkHolder.get()).thenReturn(CURATOR);
+        final ZooKeeperLockFactory zkLockFactory = new ZooKeeperLockFactory(zkHolder);
+        final ConsumerLimitingService limitingService = new ConsumerLimitingService(zkHolder, zkLockFactory, 5);
+
+        objectMapper = new JsonConfig().jacksonObjectMapper();
+        cleaningService = new ConsumerLimitingCleaningService(zkHolder, objectMapper, limitingService);
+    }
+
+    @Before
+    public void before() throws Exception {
+        deleteConsumersData();
+
+        // "hanging" node
+        CURATOR.create().creatingParentsIfNeeded().forPath("/nakadi/consumers/connections/hanging");
```

Comment:
it wasn't clear for me that `hanging` is the name of a client. It took me some time to realise that it's the name of a client which has no consumers anymore. That's why you clean it. Changing it to `consumer_with_no_open_streams` could make if more explicit. This is just a suggestion to improve clarity.